airflow/models/dag.py

```diff
@@ -1466,6 +1466,9 @@ def fetch_callback(
             # context for the callback.
             if dag.partial:
                 tis = [ti for ti in tis if not ti.state == State.NONE]
+            # filter out removed tasks
+            tis = [ti for ti in tis if ti.state != TaskInstanceState.REMOVED]
+            print(f"tis: {tis}")
```

Comment:
We don't use print in the code; if there is a need to log something, we use `logging` package, but I don't think there is a need here. If you added it for debugging purposes, this comment will be a reminder to remove it before merging the PR.
```suggestion
```

Reply:
It was only for debbug! I've removed it and added the test. Thanks